app_flutter/lib/widgets/status_grid.dart

```diff
@@ -32,191 +34,226 @@ class StatusGridContainer extends StatelessWidget {
-        child: GridView.builder(
-          addRepaintBoundaries: false,
-
-          /// The grid has as many rows as there are statuses. Additionally,
-          /// one row for task descriptions, and one at the bottom to show
-          /// a loader for more data.
-          itemCount: columnCount * (statuses.length + 2),
-          gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
-            crossAxisCount: columnCount,
-          ),
-          itemBuilder: (BuildContext context, int gridIndex) {
-            if (gridIndex == 0) {
-              /// The top left corner of the grid is nothing since
-              /// the left column is for [CommitBox] and the top
-              /// row is for [TaskIcon].
-              return const SizedBox();
-            }
+    return LatticeScrollView(
+      // TODO(ianh): Provide some vertical scroll physics that disable
+      // the clamping in the vertical direction, so that you can keep
+      // scrolling past the end instead of hitting a wall every time
+      // we load.
+      // TODO(ianh): Trigger the loading from the scroll offset,
+      // rather than the current hack of loading during build.
+      cells: _processCommitStatuses(widget.commitStatuses),
+      cellSize: const Size.square(TaskBox.cellSize),
+    );
+  }
 
-            /// Loader row at the bottom of the grid.
-            if (_isLastRow(
-              gridIndex: gridIndex,
-              columnCount: columnCount,
-            )) {
-              final int loaderIndex = gridIndex % columnCount;
-
-              /// This loader row will spell out [loadingText].
-              const String loaderRowText = 'LOADING ';
-              final Color lastRowColor = buildState.moreStatusesExist ? Colors.blueGrey : Colors.transparent;
-              final String keyValue =
-                  buildState.moreStatusesExist ? 'loader-$loaderIndex' : 'hidden-loader-$loaderIndex';
-
-              /// Only trigger [fetchMoreCommitStatuses] API call once for
-              /// this loading row.
-              if (loaderIndex == 0 && buildState.moreStatusesExist) {
-                buildState.fetchMoreCommitStatuses();
-              }
-
-              return Container(
-                key: insertCellKeys ? Key(keyValue) : null,
-                color: lastRowColor,
-                child: Padding(
-                  padding: const EdgeInsets.fromLTRB(10.0, 14.0, 10.0, 14.0),
-                  child: buildState.moreStatusesExist
-                      ? Text(loaderRowText[loaderIndex % loaderRowText.length],
-                          style: const TextStyle(
-                            color: Colors.white,
-                            fontSize: 20,
-                          ))
-                      : null,
-                ),
-              );
+  List<List<LatticeCell>> _processCommitStatuses(List<CommitStatus> commitStatuses) {
```

Comment:
Can you add the steps for converting List<CommitStatus> to the 2D matrix? It was in `TaskMatrix` and helps outline what this is doing.

Reply:
Added docs.